From 982c6ce6720bca35524a1f06d0c1a64afd77e918 Mon Sep 17 00:00:00 2001
From: Mehdi Sabwat <mehdi@videolabs.io>
Date: Thu, 3 Jun 2021 22:50:44 +0200
Subject: [PATCH 1/1] contrib: set RANLIB for toolchain.cmake

This commit will allow taking into account $RANLIB when we are cross compiling contribs that use cmake.

When cross compiling Openjpeg with emscripten sdk, for example, the build system
won't be able to use emranlib (ranlib targetting the $HOST).

cf: https://emscripten.org/docs/compiling/Building-Projects.html#troubleshooting
---
 contrib/src/main.mak | 1 +
 1 file changed, 1 insertion(+)

diff --git a/contrib/src/main.mak b/contrib/src/main.mak
index fb71479d68..2d4bce03ee 100644
--- a/contrib/src/main.mak
+++ b/contrib/src/main.mak
@@ -597,6 +597,7 @@ else
 endif
 endif
 	echo "set(CMAKE_AR $(AR) CACHE FILEPATH \"Archiver\")" >> $@
+	echo "set(CMAKE_RANLIB $(RANLIB) CACHE FILEPATH \"Add index to Archive\")" >> $@
 ifdef HAVE_CROSS_COMPILE
 	echo "set(_CMAKE_TOOLCHAIN_PREFIX $(HOST)-)" >> $@
 ifdef HAVE_ANDROID
-- 
2.32.0

